pkg/controller/beat/common/config.go

```diff
@@ -109,6 +110,13 @@ func buildBeatConfig(
 		return nil, err
 	}
 
+	// if metrics monitoring is enabled, then enable the metrics http endpoint for metricsbeat sidecar.
+	if monitoring.IsMetricsDefined(&params.Beat) {
+		if err = cfg.MergeWith(settings.MustCanonicalConfig(map[string]bool{"http.enabled": true})); err != nil {
```

Comment:
Don't we also need to disable (self)monitoring and set a UUID according to this doc https://www.elastic.co/guide/en/beats/metricbeat/current/monitoring-metricbeat-collection.html

Reply:
disabled self monitoring, and no UUID needed, as I testing with/without.